julia-assembly/src/mips32_plot.S

```diff
@@ -43,21 +44,18 @@ buffer_ini:
 
 # Going to need a register to store the buff location
 .macro write_buffer (%data, %len)
-    bgt $t2, BUF_SZ, flush_buffer # Store sp location in stack or somewhere so write can jump back
-    sw $data, 0($t1)
-    li $t1, $len
-    addi $t2, $t2, $len
+    blt $t2, BUF_SZ, not_flush_buffer # Store sp location in stack or somewhere so write can jump back
```

Comment:
Labels are allowed inside macros (see full commit message). No need to store $pc in stack